crates/storage/provider/src/utils.rs

```diff
@@ -24,6 +24,11 @@ pub fn insert_block<'a, TX: DbTxMut<'a> + DbTx<'a>>(
     // Put header with canonical hashes.
     tx.put::<tables::Headers>(block_num_hash, block.header.as_ref().clone())?;
     tx.put::<tables::HeaderNumbers>(block.hash(), block.number)?;
+    tx.put::<tables::HeaderTD>(
+        block_num_hash,
+        if has_block_reward { U256::ZERO } else { U256::from(58_750_000_000_000_000_000_000u128) }
+            .into(),
+    )?;
```

Comment:
Just to make sure I understand:
* If it has block reward, it means it's pre-merge, as a result the TotalDifficulty stage will populate that value and we can leave it to 0
* If it has no block reward, it's post merge, so TD doesn't go up anymore, so it's at `58_750_000_000_000_000_000_000u128`

Should we have a parameter for the max block the TD stage should be running until, optionally, for merged networks?